### PR TITLE
[Supernova] Silence Scheduler for Supernova

### LINKED ISSUE #398: [Supernova] add "Scheduled Maintenance" feature for creating predefined and prescheduled silences

### PR DESCRIPTION
**Pull request by TilmanHaupt**
Add a form to schedule silences for e.g. maintenance. 

fixes #398

### CODE REVIEW

File: apps/exampleapp/src/components/AppContent.js [hunk 1]
```diff
@@ -53,6 +56,20 @@ const AppContent = (props) => {
...
           <Container py>
+            <DateTimePicker
+              value={testState?.date?.end}
+              dateFormat="Y-m-d H:i"
+              label="Select a end date"
+              enableTime
+              time_24hr
+              required
+              onChange={(e) => {
+                console.log(">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>end", e)
```
Comment (ArtieReus) [inline apps/exampleapp/src/components/AppContent.js:67]: Please remove comments

File: apps/exampleapp/src/components/AppContent.js [hunk 2]
```diff
@@ -53,6 +56,20 @@ const AppContent = (props) => {
...
+                setTestState({ ...testState, date: { end: e } })
+
+                console.log(testState)
```
Comment (ArtieReus) [inline apps/exampleapp/src/components/AppContent.js:70]: same same

File: apps/supernova/src/AppContent.jsx [hunk 3]
```diff
@@ -21,7 +21,7 @@ import AlertDetail from "./components/alerts/AlertDetail"
 import PredefinedFilters from "./components/filters/PredefinedFilters"
 
 const AppContent = (props) => {
-  const { addMessage } = useActions()
+  const { addMessage, resetMessages } = useActions()
```
Comment (ArtieReus) [inline apps/supernova/src/AppContent.jsx:24]: do you really use resetMessages?

File: apps/supernova/src/components/silences/SilenceScheduled.jsx [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+  const apiEndpoint = useGlobalsApiEndpoint()
+
+  // set sucess of sending the silence
+  const [success, setSuccess] = useState(null)
+
+  // set the selected template
+  const [selected, setSelected] = useState(null)
+
+  // default time for DateTimePicker
+  const date = new Date()
```
Comment (ArtieReus) [inline apps/supernova/src/components/silences/SilenceScheduled.jsx:43]: try you memorize the default data. Each rendering would then create a new date.

File: apps/supernova/src/components/silences/silenceScheduledHelpers.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    formState = {
+      ...formState,
+      comment: {
+        ...formState.comment, // Only create the comment object if it exists
+        error: "Please enter at least 3 characters",
+      },
+    }
+  }
+
+  console.log(formState.date.start)
```
Comment (ArtieReus) [inline apps/supernova/src/components/silences/silenceScheduledHelpers.js:36]: same same

**Review by ArtieReus — CHANGES_REQUESTED** [review]

**Comment by ArtieReus** [comment]
When opening the Template form without selecting any template you can still press save and an error will be displayed. I will propose to disable the save button until the template is selected. What do you think?

**Comment by ArtieReus** [comment]
If the template have same title the select displays the two templates selected. This is probably because the key is the title. I would recomend to create unique ids (index) to avoid it.

**Comment by ArtieReus** [comment]
Check how the key status is used in the old Supernova. Ex: 
```
"status": "active",
```

**Comment by ArtieReus** [comment]
Validate template input. Changing for example `editable_labels ` as string crashes the component.

File: apps/supernova/src/components/silences/silenceScheduledHelpers.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    start: null,
+    end: null,
+    error: null,
+  },
+}
+
+////// Form Validation
+
+export const validateForm = (b) => {
+  console.log(b.date.start)
```
Comment (edda) [inline apps/supernova/src/components/silences/silenceScheduledHelpers.js:19]: console.log remove plx

File: apps/supernova/src/components/silences/silenceScheduledHelpers.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  let formState = b
+  console.log(formState.date.start)
```
Comment (edda) [inline apps/supernova/src/components/silences/silenceScheduledHelpers.js:21]: and again :)

File: apps/supernova/src/components/silences/silenceScheduledHelpers.js (hunk 5 above)
Comment (edda) [inline apps/supernova/src/components/silences/silenceScheduledHelpers.js:36]: one more

**Review by edda — CHANGES_REQUESTED** [review]

File: apps/exampleapp/src/components/AppContent.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (TilmanHaupt) [reply apps/exampleapp/src/components/AppContent.js:67]: komplettes file ist rm

**Review by ArtieReus — APPROVED** [review]
Good Job!! lgtm!

**Pull request merged**